src/box/lua/upgrade.lua

```diff
@@ -2125,6 +2126,10 @@ local function downgrade_impl(version_str, dry_run)
     end
 
     local schema_version_cur = get_version()
+    local app_version = tarantool.version:match('^%d+%.%d+%.%d+')
```

Comment:
Discussed f2f and decided to keep this way. The issue is otherwise we need to somehow convert `tarantool_version_id` value to string to report application version in the error message in the diff below.